File: soln-heap-problems/kth-largest-element-in-an-array-with-min-heap.java
/*
https://leetcode.com/problems/kth-largest-element-in-an-array/
*/

class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap=new PriorityQueue<>(k);
        int i=0;
        while(i<nums.length){
            minHeap.offer(nums[i]);
            if(minHeap.size()>k)
                minHeap.poll();
            i++;
        }
        return minHeap.peek();        
    }
}

File: C++/soln-heap-problems/kth-largest-element-in-an-array-with-min-heap.cpp
/**
 @author Farheen Bano
  
 Date 19-07-2021
 
 Reference-
 https://leetcode.com/problems/kth-largest-element-in-an-array/
*/

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> minHeap;
        int i=0;
        while(i<nums.size()){
            minHeap.push(nums[i]);
            if(minHeap.size()>k)
                minHeap.pop();
            i++;
        }
        return minHeap.top();
    }
};